src/Perspex.SceneGraph/Rect.cs

```diff
@@ -297,7 +297,8 @@ public override bool Equals(object obj)
         {
             if (obj is Rect)
             {
-                return this == (Rect)obj;
+                var other = (Rect)obj;
+                return Position == other.Position && Size == other.Size;
```

Comment:
To be the same as for Point:
https://github.com/wieslawsoltes/Perspex/blob/ci/src/Perspex.SceneGraph/Point.cs#L202